### DISCUSSION TITLE
Create a visual graph of atom dependencies

### DISCUSSION THREAD

**Discussion by axelson** [pmndrs/jotai#1056] [Q&A] [answered]
Is it possible to create a graph of atom dependencies in something like [graphviz](https://en.wikipedia.org/wiki/Graphviz)? We're tracking down some performance issues and have enough atoms that it's difficult to keep track of what atoms are being depended on by which atoms/components.

**Comment by dai-shi** [comment, marked as answer]
Good point. We expose dependencies in Provider's debug state and `useAtomsDevtools`, but they can't be exported as reusable data.

#911 is an issue to improve it:
> Maybe we should somehow return dependents too. (Provider's useDebugState does it)

**Reply by axelson** [reply]
Ah, thanks for the link. I'll watch that issue.
